ipn/ipnlocal/local.go

```diff
@@ -2097,8 +2097,15 @@ func (b *LocalBackend) shouldUploadServices() bool {
 
 func (b *LocalBackend) SetCurrentUserID(uid string) {
 	b.mu.Lock()
-	b.pm.SetCurrentUser(uid)
-	b.mu.Unlock()
+	if b.pm.CurrentUser() == uid {
+		b.mu.Unlock()
+		return
+	}
+	if err := b.pm.SetCurrentUser(uid); err != nil {
+		b.mu.Unlock()
+		return
+	}
+	b.resetForProfileChangeLockedOnEntry()
```

Comment:
oh no, more "LockedOnEntry"? I wanted to eliminate the one we had :(

I'll approve, but that sucks

Reply:
I agree, I plan on refactoring this package next.